src/sources/journald.rs

```diff
@@ -80,13 +83,42 @@ pub struct JournaldConfig {
     remap_priority: bool,
 }
 
+impl JournaldConfig {
+    fn merged_include_matches(&self) -> crate::Result<Matches> {
+        let include_units = match (!self.units.is_empty(), !self.include_units.is_empty()) {
+            (true, true) => return Err(BuildError::BothUnitsAndIncludeUnits.into()),
+            (true, false) => {
+                warn!("The `units` setting is deprecated, use `include_units` instead.");
+                &self.units
+            }
+            (false, _) => &self.include_units,
+        };
+
+        Ok(Self::merge_units(&self.include_matches, include_units))
+    }
+
+    fn merged_exclude_matches(&self) -> Matches {
+        Self::merge_units(&self.exclude_matches, &self.exclude_units)
+    }
+
+    fn merge_units(matches: &Matches, units: &[String]) -> Matches {
+        let mut matches = matches.clone();
+        units.iter().map(|s| fixup_unit(s)).for_each(|unit| {
+            let entry = matches.entry(String::from(SYSTEMD_UNIT));
+            entry.or_default().insert(unit);
+        });
```

Comment:
I don't think this does what you want. Rust iterators are lazy, so they don't evaluate anything unless you extract a value out (like `.collect()`). I would think clippy would warn about this, so I'm a little puzzled. In any case, I think a straight loop would be clearer here:
```suggestion
        for unit in units {
            let entry = matches.entry(String::from(SYSTEMD_UNIT));
            entry.or_default().insert(fixup_unit(unit));
        }
```

Reply:
In the case of for_each, it works as expected.
But, the for syntax you suggested is simpler, so I fixed it with ea3c0de